src/ansiblelint/schemas/ansible-lint-config.json

```diff
@@ -122,60 +122,7 @@
         "$ref": "#/$defs/rule"
       },
       "propertyNames": {
-        "enum": [
-          "command-instead-of-module",
-          "command-instead-of-shell",
-          "deprecated-bare-vars",
-          "deprecated-command-syntax",
-          "deprecated-local-action",
-          "deprecated-module",
-          "empty-string-compare",
-          "fqcn",
-          "fqcn[action-core]",
-          "fqcn[action-redirect]",
-          "fqcn[action]",
-          "galaxy",
-          "galaxy[no-changelog]",
-          "galaxy[tags]",
-          "galaxy[version-incorrect]",
-          "galaxy[version-missing]",
-          "ignore-errors",
-          "inline-env-var",
-          "internal-error",
-          "jinja",
-          "key-order",
-          "latest",
-          "literal-compare",
-          "load-failure",
-          "meta-incorrect",
-          "meta-no-info",
-          "meta-no-tags",
-          "meta-runtime",
-          "meta-video-links",
-          "name",
-          "no-changed-when",
-          "no-handler",
-          "no-jinja-when",
-          "no-log-password",
-          "loop-var-prefix",
-          "no-prompting",
-          "no-relative-paths",
-          "no-same-owner",
-          "no-tabs",
-          "only-builtins",
-          "package-latest",
-          "parser-error",
-          "partial-become",
-          "playbook-extension",
-          "risky-file-permissions",
-          "risky-octal",
-          "risky-shell-pipe",
-          "role-name",
-          "schema",
-          "syntax-check",
-          "var-naming",
-          "yaml"
-        ]
+        "pattern": "^[a-z0-9-\\[\\]]+$"
```

Comment:
No current rule name that I'm aware of contains numbers, so why include these. On the other hand, underscores and capital letters might be something custom rule authors want to use?

Reply:
I think rule names (IDs) and file names should be standardized, but I don't have a strong opinion on how these should be.  Also, IMHO, it's not a point of discussion for this issue. We can update the pattern after the standard is fixed.

For now, the pattern is tentatively specified based on the IDs of the standard rules and the IDs of some custom rules under test/custom_rules/.